src/os/ui/featureedit.js

```diff
@@ -161,6 +161,18 @@ os.ui.FeatureEditCtrl = function($scope, $element, $timeout) {
    */
   this['size'] = os.style.DEFAULT_FEATURE_SIZE;
 
+  /**
+   * The feature fill color
+   * @type {string}
+   */
+  this['fillColor'] = os.style.DEFAULT_LAYER_COLOR;
```

Comment:
This essentially causes the color issue I found (where the fill color doesn't stay in sync with the stroke color for new features). Basically `os.style.DEFAULT_LAYER_COLOR` is not a hex string, but everywhere that `fillColor` is used expects it that way, including the color picker. The `labelColor` doesn't have that problem because the initial setting gets converted to hex before going to the color picker, while the `fillColor` is only changed when `loadFromFeature()` is called (so `createPreviewFeature()` addresses `labelColor` but not `fillColor` in line 924).

```
this['fillColor'] = os.color.toHexString(os.style.DEFAULT_LAYER_COLOR);
```

With that being said, it would also be wise to initialize `color` and `labelColor` this way too.

Reply:
I added the logic to convert the globals to hex strings to color, labelColor, and fillColor.